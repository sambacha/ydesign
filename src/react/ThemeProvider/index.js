import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
