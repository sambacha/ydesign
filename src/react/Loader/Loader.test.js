import Loader from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Loader component sanity', () => {
  it('has name', () => {
    expect(Loader.displayName).toBe('Loader');
  });

  it('matches default snapshot', () => {
    const component = render(<Loader />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Loader />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
