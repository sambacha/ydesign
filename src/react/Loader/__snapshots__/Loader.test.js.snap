// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #4E3FCE;
  background-color: transparent;
  width: 1rem;
  height: 1rem;
}

.c1 {
  display: block;
  position: relative;
  overflow: hidden;
}

.c1 > svg {
  fill: currentColor;
  height: 100%;
  width: 100%;
  display: block;
  -webkit-animation: hseeoX 1000ms linear infinite;
  animation: hseeoX 1000ms linear infinite;
  -webkit-animation-timing-function: cubic-bezier(0.445,0.05,0.55,0.95);
  animation-timing-function: cubic-bezier(0.445,0.05,0.55,0.95);
}

<div
  class="c0 c1"
  color="primary"
>
  <svg
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M12 20.5714C16.7339 20.5714 20.5714 16.7339 20.5714 12C20.5714 7.26613 16.7339 3.42857 12 3.42857C7.26613 3.42857 3.42857 7.26613 3.42857 12C3.42857 16.7339 7.26613 20.5714 12 20.5714ZM12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
      fill="currentColor"
      fill-rule="evenodd"
      opacity="0.25"
    />
    <mask
      height="24"
      id="mask0"
      mask-type="alpha"
      maskUnits="userSpaceOnUse"
      width="24"
      x="0"
      y="0"
    >
      <path
        clip-rule="evenodd"
        d="M12 20.5714C16.7339 20.5714 20.5714 16.7339 20.5714 12C20.5714 7.26613 16.7339 3.42857 12 3.42857C7.26613 3.42857 3.42857 7.26613 3.42857 12C3.42857 16.7339 7.26613 20.5714 12 20.5714ZM12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask0)"
    >
      <path
        d="M12 24V15.8824V12H24V24H12Z"
        fill="currentColor"
      />
    </g>
  </svg>
</div>
`;

exports[`Loader component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #4E3FCE;
  background-color: transparent;
  width: 1rem;
  height: 1rem;
}

.c1 {
  display: block;
  position: relative;
  overflow: hidden;
}

.c1 > svg {
  fill: currentColor;
  height: 100%;
  width: 100%;
  display: block;
  -webkit-animation: hseeoX 1000ms linear infinite;
  animation: hseeoX 1000ms linear infinite;
  -webkit-animation-timing-function: cubic-bezier(0.445,0.05,0.55,0.95);
  animation-timing-function: cubic-bezier(0.445,0.05,0.55,0.95);
}

<div
  class="c0 c1"
  color="primary"
>
  <svg
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M12 20.5714C16.7339 20.5714 20.5714 16.7339 20.5714 12C20.5714 7.26613 16.7339 3.42857 12 3.42857C7.26613 3.42857 3.42857 7.26613 3.42857 12C3.42857 16.7339 7.26613 20.5714 12 20.5714ZM12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
      fill="currentColor"
      fill-rule="evenodd"
      opacity="0.25"
    />
    <mask
      height="24"
      id="mask0"
      mask-type="alpha"
      maskUnits="userSpaceOnUse"
      width="24"
      x="0"
      y="0"
    >
      <path
        clip-rule="evenodd"
        d="M12 20.5714C16.7339 20.5714 20.5714 16.7339 20.5714 12C20.5714 7.26613 16.7339 3.42857 12 3.42857C7.26613 3.42857 3.42857 7.26613 3.42857 12C3.42857 16.7339 7.26613 20.5714 12 20.5714ZM12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask0)"
    >
      <path
        d="M12 24V15.8824V12H24V24H12Z"
        fill="currentColor"
      />
    </g>
  </svg>
</div>
`;
